src/pages/ArticleList.jsx

```diff
@@ -2,22 +2,32 @@ import { useEffect, useState } from "react";
 import newsAPIGet from "../utils/utils";
 import ArticleCard from "../components/ArticleCard";
 import { useParams } from "react-router-dom";
+import SortArticles from "../components/SortArticles";
 
 export default function ArticleList() {
 	const [articlesList, setArticlesList] = useState([]);
 	const [isLoading, setIsLoading] = useState(false);
 	const [errorState, setErrorState] = useState(null);
 
 	const { topicSlug } = useParams();
+	const [sortBy, setSortBy] = useState("");
+	const [sortOrder, setSortOrder] = useState("desc");
 
 	useEffect(() => {
-		let url = "/articles";
+		const url = "/articles";
+		const apiQuery = new URLSearchParams();
 		if (topicSlug) {
-			console.log(topicSlug);
-			url += `/?topic=${topicSlug}`;
+			apiQuery.append("topic", topicSlug);
 		}
+		if (sortBy) {
+			apiQuery.append("sort_by", sortBy);
+		}
+		if (sortOrder) {
+			apiQuery.append("order", sortOrder);
+		}
+
 		setIsLoading(true);
-		newsAPIGet(url)
+		newsAPIGet(`${url}?${apiQuery}`)
```

Comment:
You could make sure of axios' ability to handle optional params here, if you pass through the values for the sorting and topics etc and if they're undefined axios will simply ignore them. Take a look at the axios notes for help 👍

Reply:
Great tip - that's much cleaner :smile: